src/vs/workbench/api/node/extHostDebugService.ts

```diff
@@ -917,6 +917,12 @@ export class ExtHostDebugSession implements vscode.DebugSession {
 		return this._name;
 	}
 
+	public set name(name: string) {
+		this._name = name;
+		this._configuration.name = name;
```

Comment:
I do not like that we also change the `this._configuration.name`
More about this in a comment below

Reply:
I agree: `configuration.name` is used as the initial name for the session but becomes decoupled from the session's name as soon as the session's name is changed.

Reply:
yes, don't touch `this._configuration.name`

Reply:
Done.